v2/outbound_transport_middleware.go

```diff
@@ -75,6 +75,11 @@ func applyUnaryOutboundTransportMiddleware(o UnaryOutbound, f UnaryOutboundTrans
 // UnaryOutboundTransportMiddlewareFunc adapts a function into a UnaryOutboundTransportMiddleware.
 type UnaryOutboundTransportMiddlewareFunc func(context.Context, *Request, *Buffer, UnaryOutbound) (*Response, *Buffer, error)
 
+// Name for UnaryOutboundTransportMiddlewareFunc.
+func (UnaryOutboundTransportMiddlewareFunc) Name() string {
+	return "UnaryOutboundTransportMiddlewareFunc"
```

Comment:
This might be problematic if we intend users to reference this to implement the `UnaryOutboundTransportMiddleware`. If two middleware were provided into the container using this func then their would be a name registration conflict, for example.

Reply:
Yeh, thread a custom name through the constructor.

Reply:
🤔 good point

I'll have to make a new type which holds a middleware func and a name field since there is no constructor